server/aesdsocket.c

```diff
@@ -0,0 +1,247 @@
+        	return -1;
+    	}
+    		
+	freeaddrinfo(server_info);
+	addr_size = sizeof Client_addr;
+	
+	while(1){
+	
+	        int count = 0;
+        	int recv_buff_len = 0;
+        	int prev_len = TEMP_BUFFER_SIZE;
+        	bool packet_received = false;
+		status=0;
+		
+		//accept
+		client_socket_fd= accept(server_socket_fd, (struct sockaddr *)&Client_addr, &addr_size);
+	 	if(client_socket_fd == -1 )
+	 	{
+            		syslog(LOG_ERR, "Client connection failed");
+            		return -1;
+        	}
+        
+		//for client_ip
+               client_ip = inet_ntop(AF_INET, &Client_addr.sin_addr, address_string, sizeof(address_string));
+        	syslog(LOG_DEBUG, "Connected with %s\n\r",client_ip );
+        	
+		memset(temp_buffer, '\0', TEMP_BUFFER_SIZE);
+
+        	while( false == packet_received )
+        	{
+            		int i=0;
+            		packet_size = 0;
+            		            			
+            		if(-1 == recv(client_socket_fd, &temp_buffer, TEMP_BUFFER_SIZE, 0)){
+                		syslog(LOG_ERR, "recv failed");
+            		}
+      		      		
+            		while(i< TEMP_BUFFER_SIZE)
+            		{
+		                packet_size++;
+               		 if( '\n' == temp_buffer[i] )
+               		 {
+               		     packet_received = true;
+               		     break;
+               		 }
+               		 i++;
+            		}
+
+            		if(count != 0)
+            		{
+                		char *ptr = realloc(recieved_packets, prev_len+packet_size);
+                		if(ptr != NULL)
+                		{
+                 		 	recieved_packets = ptr;
+                  		 	prev_len += packet_size;                    			
+                		}
+                		else
+                		{
+                			syslog(LOG_ERR, "realloc failed");
+
+                		}
+             			recv_buff_len = prev_len;
+             			
+            		}
+            		else
+            		{
+            		        recieved_packets = (char *)malloc(packet_size);
+                		if(NULL == recieved_packets)
+                		{
+                    			syslog(LOG_ERR, "malloc failed");
+                		}
+                		
+          			memset(recieved_packets, '\0', packet_size);
+                		recv_buff_len = packet_size;
+
+            		}
+            		
+        		memcpy((count * TEMP_BUFFER_SIZE) + recieved_packets, temp_buffer, packet_size);
+        		count++;
+        		}
+
+        		lseek(aesdsocketdata_fd, 0, SEEK_END);
+        		
+        		status= write(aesdsocketdata_fd, recieved_packets, recv_buff_len);
+        		if( -1 == status)
+        		{
+        		    syslog(LOG_ERR, "write failed");
+        		    printf("write failed\n");
+        		}
+        		file_size += status;
+        		free(recieved_packets);
+        		memset(temp_buffer, '\0', TEMP_BUFFER_SIZE);
+        		lseek(aesdsocketdata_fd, 0, SEEK_SET);
+
+        		char *recv_buffer = (char *)malloc(file_size);
```

Comment:
@akshithaluguri22 what assumption are you making here about the total size of all packets sent and the size of the process heap?